test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -1166,15 +1168,69 @@ public async Task NoTypeLoading()
         };
 
         using var _ = CreateTempPool(builder, out var connectionString);
-        using var conn = await OpenConnectionAsync(connectionString);
-        // Arrays should not be supported in this mode
-        Assert.That(async () => await conn.ExecuteScalarAsync("SELECT '{1,2,3}'::INTEGER[]"),
-            Throws.Exception.TypeOf<NotSupportedException>());
+        await using var conn = await OpenConnectionAsync(connectionString);
         // Test that some basic types do work
         Assert.That(await conn.ExecuteScalarAsync("SELECT 8"), Is.EqualTo(8));
         Assert.That(await conn.ExecuteScalarAsync("SELECT 'foo'"), Is.EqualTo("foo"));
         Assert.That(await conn.ExecuteScalarAsync("SELECT TRUE"), Is.EqualTo(true));
         Assert.That(await conn.ExecuteScalarAsync("SELECT INET '192.168.1.1'"), Is.EqualTo(IPAddress.Parse("192.168.1.1")));
+        //The following types with hardcoded oids should be supported in this mode:
+        // Arrays
+        Assert.That(await conn.ExecuteScalarAsync("SELECT '{1,2,3}'::INTEGER[]"), Is.EquivalentTo(new []{1,2,3}));
+        // Ranges
+        Assert.That(await conn.ExecuteScalarAsync("SELECT '[3,7)'::int4range"), Is.EqualTo(new NpgsqlRange<int>(3, true, false, 7, false, false)));
+        // Arrays of ranges
+        Assert.That(await conn.ExecuteScalarAsync("SELECT '{\"[3,7)\", \"[8,)\"}'::int4range[]"),
+            Is.EquivalentTo(new[] { new NpgsqlRange<int>(3, true, false, 7, false, false), new NpgsqlRange<int>(8, true, false, 0, false, true) }));
+        // Multiranges
+        if (conn.PostgreSqlVersion >= new Version(14, 0))
+            Assert.That(await conn.ExecuteScalarAsync("SELECT '{[3,7), [8,)}'::int4multirange"),
+                Is.EquivalentTo(new[]
+                    { new NpgsqlRange<int>(3, true, false, 7, false, false), new NpgsqlRange<int>(8, true, false, 0, false, true) }));
+    }
+
+    [Test, Description("Makes sure that PostgresMinimalDatabaseInfo and PostgresDatabaseInfo are equal for freshly created databases")]
+    public async Task NoTypeLoadingVsTypeLoading()
+    {
+        if (MultiplexingMode == MultiplexingMode.Multiplexing)
+            return;
+
+        await using var conn = await OpenConnectionAsync(ConnectionString);
+        MinimumPgVersion(conn, "14.0", "Current version for generation of PostgresMinimalDatabaseInfo is 14");
+        var databaseName = Guid.NewGuid().ToString("N");
+        try
+        {
+            await conn.ExecuteNonQueryAsync($"CREATE DATABASE \"{databaseName}\" TEMPLATE template0;");
+            var loadedCsb = new NpgsqlConnectionStringBuilder(ConnectionString) { Database = databaseName };
+            using var loadedPool = CreateTempPool(loadedCsb, out var loadedCs);
```

Comment:
You should be able to use the new data source APIs here - we should in generate stop using CreateTempPool and convert all tests (opened #4572 to track this).